csq.c

```diff
@@ -3328,7 +3330,7 @@ static const char *usage(void)
         "       --targets-overlap 0|1|2       Include if POS in the region (0), record overlaps (1), variant overlaps (2) [0]\n"
         "       --threads INT                 Use multithreading with <int> worker threads [0]\n"
         "   -v, --verbose INT                 Verbosity level 0-2 [1]\n"
-        "       --write-index                 Automatically index the output files [off]\n"
+        "       --write-index[=FMT]           Automatically index the output files [off]\n"
```

Comment:
Please add an example, something like
```
--write-index[=FMT]           Automatically index the output files, FMT={csi|tbi} [off]
```

Reply:
Maybe a short option `-W, --write-index` could be added at the same time as well? (Requested here https://github.com/samtools/bcftools/issues/2139)

Reply:
I'm wary of padding the usage lines even longer.  They already don't fit on a standard 80 column window.  I think adding to documentation is fine though.

Reply:
Short options are a nightmare when it comes to optional arguments.  I mentioned this in my comment before.

I think if we want to permit -W, then we should change how we handle the optional bit as it's just a recipe for confused users frankly.

A case in point, `-W tbi` is an error, `-W=tbi` is an error, but `-Wtbi` works.  With long options it's obvious.

We could perhaps change the parsing so "=tbi" and "tbi" are both recognised as index formats, along with "=csi" and "csi", but it is a non-standard approach to optional arguments.